File: tl/src/main/java/com/github/badoualy/telegram/tl/api/TLInputPeerNotifyEventsEmpty.java
package com.github.badoualy.telegram.tl.api;

/**
 * @author Yannick Badoual yann.badoual@gmail.com
 * @see <a href="http://github.com/badoualy/kotlogram">http://github.com/badoualy/kotlogram</a>
 */
public class TLInputPeerNotifyEventsEmpty extends TLAbsInputPeerNotifyEvents {
    public static final int CONSTRUCTOR_ID = 0xf03064d8;

    public TLInputPeerNotifyEventsEmpty() {
    }

    @Override
    public String toString() {
        return "inputPeerNotifyEventsEmpty#f03064d8";
    }

    @Override
    public int getConstructorId() {
        return CONSTRUCTOR_ID;
    }

    @Override
    @SuppressWarnings("PointlessBooleanExpression")
    public boolean equals(Object object) {
        if (!(object instanceof TLInputPeerNotifyEventsEmpty)) return false;
        if (object == this) return true;

        TLInputPeerNotifyEventsEmpty o = (TLInputPeerNotifyEventsEmpty) object;

        return true;
    }

    @Override
    public final boolean isEmpty() {
        return true;
    }

    @Override
    public final boolean isNotEmpty() {
        return false;
    }
}

File: tl/src/main/kotlin/com/github/badoualy/telegram/tl/api/TLInputPeerNotifyEventsEmpty.kt
package com.github.badoualy.telegram.tl.api

/**
 * inputPeerNotifyEventsEmpty#f03064d8
 *
 * @author Yannick Badoual yann.badoual@gmail.com
 * @see <a href="http://github.com/badoualy/kotlogram">http://github.com/badoualy/kotlogram</a>
 */
class TLInputPeerNotifyEventsEmpty : TLAbsInputPeerNotifyEvents() {
    private val _constructor: String = "inputPeerNotifyEventsEmpty#f03064d8"

    override val constructorId: Int = CONSTRUCTOR_ID

    override fun toString() = _constructor

    override fun equals(other: Any?): Boolean {
        if (other !is TLInputPeerNotifyEventsEmpty) return false
        if (other === this) return true

        return true
    }
    companion object  {
        const val CONSTRUCTOR_ID: Int = 0xf03064d8.toInt()
    }
}
